spot_driver/spot_driver/spot_ros2.py

```diff
@@ -2062,36 +2079,63 @@ def _get_robot_command_feedback(self, goal_id: Optional[str]) -> RobotCommandFee
         return feedback
 
     def handle_robot_command(self, goal_handle: ServerGoalHandle) -> RobotCommand.Result:
+        """
+        Spot cannot process long trajectories. If we issue a command with long
+        trajectories for the arm or the body, the command will be batched,
+        assuming that there is only one long trajectory or more but all time
+        aligned.
+        To account for the network latency, a command must contain batched
+        trajectory with some overlapping.
+        This is currently not possible for the gripper trajectory, due to some
+        limitation in the SDK.
+        """
+        if self.spot_wrapper is None:
+            return
+
         ros_command = goal_handle.request.command
         proto_command = robot_command_pb2.RobotCommand()
         convert(ros_command, proto_command)
-        self._wait_for_goal = None
-        if self.spot_wrapper is None:
-            self._wait_for_goal = WaitForGoal(self.get_clock(), 2.0)
-            goal_id = None
-        else:
-            success, err_msg, goal_id = self.spot_wrapper.robot_command(proto_command)
-            if not success:
-                raise Exception(err_msg)
 
-        self.get_logger().info("Robot now executing goal " + str(goal_id))
-        # The command is non-blocking, but we need to keep this function up in order to interrupt if a
-        # preempt is requested and to return success if/when the robot reaches the goal. Also check the is_active to
-        # monitor whether the timeout_cb has already aborted the command
+        # Inspect the command and if there are long trajectories, batch them.
+        commands = robot_command_util.batch_command(
+            proto_command, SpotROS.TRAJECTORY_BATCH_SIZE, SpotROS.TRAJECTORY_BATCH_OVERLAPPING_POINTS
+        )
+        num_of_commands = len(commands)
+
+        goal_id = None
+        self._wait_for_goal = None
         feedback: Optional[RobotCommandFeedback] = None
         feedback_msg: Optional[RobotCommand.Feedback] = None
+
+        start_time = time.time()
+        time_to_send_command = 0.0
+
+        index = 0
         while (
             rclpy.ok()
+            and goal_handle.is_active
```

Comment:
I'm a little confused how this while loop keeps going until the commands are done sending.  Could you add a comment?

Reply:
It basically keeps going until one of the trajectory command in the batch succeeds i.e. reaches the goal. Because we send the next batch before the previous one is completed, the only batch that succeeds is the last to be sent.
I will add a comment in the code.